lib/system/seqs_v2.nim

```diff
@@ -93,7 +104,13 @@ proc grow*[T](x: var seq[T]; newLen: Natural; value: T) =
     xu.p = cast[typeof(xu.p)](prepareSeqAdd(oldLen, xu.p, newLen - oldLen, sizeof(T), alignof(T)))
   xu.len = newLen
   for i in oldLen .. newLen-1:
-    xu.p.data[i] = value
+    when supportsCopyMem(T):
+      # no copy hook exists, a direct assignment can be used
+      xu.p.data[i] = value
+    else:
+      # the slot is in an unknown state, so a direct assignment (which would
+      # invoke the copy hook) cannot be used
+      rawAssign(xu.p.data[i], value)
```

Comment:
I think introducing a magic like the following would make sense:
```nim
proc unknown[T](x: var T): var T {.magic: "Unknown".}
```

What it'd do is mark the argument location as being in an unknown/uninitialized state, preventing the compiler from expanding an assignment into a hook call. It's usage would look like this: `unknown(xu.p.data[i]) = value`.

The idea is to provide a more precise tool than `.nodestroy` for safely and efficiently implementing low-level facilities where destructor-having types need to be assigned to some uninitialized memory location.

In this specific case, the `rawAssign` workaround would become obsolete.

Reply:
Good idea, for name, perhaps `asUninitialized` is a tad more descriptive and since it likely won't be used often, the extra length seems acceptable.